tqdm/std.py

```diff
@@ -126,6 +127,38 @@ def create_th_lock(cls):
 TqdmDefaultWriteLock.create_th_lock()
 
 
+class CallbackIOWrapper(object):
+    def __getattr__(self, name):
+        return getattr(self._wrapped, name)
+
+    def __setattr__(self, name, value):
+        return setattr(self._wrapped, name, value)
+
+    def __init__(self, callback, stream, method="read"):
+        """
+        Wrap a given `file`-like object's `read()` or `write()` to report
+        lengths to the given `callback`
+        """
+        object.__setattr__(self, '_wrapped', stream)
```

Comment:
@altendky just checking - do you think we need to hassle `encoding` here as with `SimpleTextIOWrapper`. I don't think we do.

Reply:
At first glance, no.  The purpose here isn't to encode but simply to let you have a chance to see how much stuff is going through.